api/src/main/resources/moduleApplicationContext.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
I'd strongly prefer is we could not use Spring here, especially because the annotation configuration add unnecessarily to this. Just collapse the `DemoConditionGenerator` and `DemoDiagnosisGenerator` into standard classes.

Reply:
I had added these to spring because of their evolving nature of the openmrs platform.

Reply:
In that case, can we switch to just using `<bean ...>` entries rather than annotation scanning? Annotation scanning tends to be a bad idea in OpenMRS in general.

Reply:
It'd not be easy to conditionally load the `<bean ..>` defined entries overriding them with others.  Also that this happens at startup, I'd think this as not being inefficient to the system's performance.

Reply:
True, but in this instance we don't have any conditionally loaded beans any more (well, technically, they are, but the conditions will always be met). Unfortunately, component scanning really does add a substantial amount of time to start-up (more specifically, this is one of the main causes of slowness during the app context refresh that happens after all modules are initially started).

Finally, unlike most modules, this module is intended to be used in context where the main "performance" cost is startup time—think starting an OpenMRS instance as part of CI job, so even if it's not a drag on overall system performance, it's a drag where we don't want it to be.